korangar/src/loaders/model/mod.rs

```diff
@@ -266,14 +267,25 @@ impl ModelLoader {
                     bounding_box.center().z,
                 ));
 
-                parent_matrix * translation_matrix * node.transform_matrix
+                translation_matrix * node.transform_matrix
             }
-            false => parent_matrix * node.transform_matrix,
+            false => node.transform_matrix,
         };
+        node.static_node_matrix = parent_matrix * node.transform_matrix;
 
-        node.child_nodes
-            .iter_mut()
-            .for_each(|child_node| Self::calculate_transformation_matrix(child_node, false, bounding_box, node.transform_matrix));
+        node.child_nodes.iter_mut().for_each(|child_node| {
+            Self::calculate_transformation_and_static_matrix(child_node, false, bounding_box, node.static_node_matrix);
+        });
+    }
+
+    // A model is static, if it doesn't have any animations.
+    pub fn is_static(node: &mut Node) -> bool {
```

Comment:
Why does this need to be mut? I don't see any mutation happening

Reply:
Copy pasta =/

Reply:
Great catch!